### PR TITLE
Cleanup flake.nix

### PR DESCRIPTION
**Pull request by gytis-ivaskevicius**
Cleaned up flake.nix

Notable changes:
- Removed `defaultPackage` attribute since it has been deprecated and replaced with `packages.default`
- Removed `apps` since its just unnecessary code, by default if app is not found `packages` attributeset is looked up
- Overlay evaluation no longer depends on `nixpkgs` flake input (that's a terrible design, overlays get `nixpkgs` passed as an argument)

### CODE REVIEW

File: flake.nix [hunk 1]
```diff
@@ -9,31 +9,28 @@
...
-      packages = forAllSystems (system: rec {
-        polymc = pkgs.${system}.libsForQt5.callPackage ./nix { inherit version self libnbtplusplus; };
-        polymc-qt6 = pkgs.${system}.qt6Packages.callPackage ./nix { inherit version self libnbtplusplus; };
-        
-        default = polymc;
-      });
 
-      defaultPackage = forAllSystems (system: self.packages.${system}.default);
+      packagesFn = pkgs: rec {
+        default = polymc;
```
Comment (zimbatm) [inline flake.nix]: This means that it would set `nixpkgs.default` if the overlay is being used.

**Review by zimbatm — COMMENTED** [review]
Overall it's better than the previous version. It would be even simpler with no overlay :)

File: flake.nix (hunk 1 above)
Comment (gytis-ivaskevicius) [reply flake.nix]: Right, I did not think about it. Fixed

**Review by Scrumplex — APPROVED** [review]
Thanks

**Review by flowln — APPROVED** [review]
thank u!

**Review by txtsd — APPROVED** [review]

**Comment by TheOPtimal** [comment]
Why was this approved and merged?

1. `defaultPackage` is still necessary for compatibility, lots of people still haven't upgraded to Nix 2.7 or later
2. Why remove `apps`? The fact that Nix falls back on the default package is just that - a fallback. It shouldn't be relied upon, and that behaviour could even change in a future version of Nix.

The only things that makes sense here are the comment edits and the overlay change.

**Comment by gytis-ivaskevicius** [comment]
Chances are that there are very few people using nix 2.7 since right now you cant even install nix 2.7 or earlier without downgrading nixpkgs by quite a bit. And it's not like there are many peeps using it from CLI. 

I can make exactly the same case for leaving apps:
Why leave apps? The fact that Nix falls back on the default package is a spec. Apps shouldn't be relied upon since they could change in a future version of Nix.

Also, have you noticed that nixpkgs has no `apps` attribute? ;)

**Comment by TheOPtimal** [comment]
> Chances are that there are very few people using nix 2.7 since right now you cant even install nix 2.7 or earlier without downgrading nixpkgs by quite a bit. And it's not like there are many peeps using it from CLI.
> 
> I can make exactly the same case for leaving apps: Why leave apps? The fact that Nix falls back on the default package is a spec. Apps shouldn't be relied upon since they could change in a future version of Nix.
> 
> Also, have you noticed that nixpkgs has no `apps` attribute? ;)

Not referring to Nix 2.7 specifically, I meant **below** 2.7

Also, if you're writing a `flake.nix`, nixpkgs is the last thing you should be looking at. `legacyPackages` had to be created just for `nixpkgs` because it's so out of line with what flakes are supposed to be.

And I don't think there even is a spec, or at least the one that exists is incomplete. Flakes are still in beta and everchanging.

Also, if you don't define `apps` properly, then the name of the derivation is relied on, which messes things up because then `nix run github:PolyMC/PolyMC#polymc-qt6` will try to run an executable named `polymc-qt6` which doesn't exist.

**Pull request merged**